extensions/resource-deployment/src/interfaces.ts

```diff
@@ -189,8 +189,8 @@ export type ComponentCSSStyles = {
 export interface IOptionsSource {
 	readonly type: OptionsSourceType,
 	readonly variableNames: { [index: string]: string; },
-	getOptions(): Promise<string[] | azdata.CategoryValue[]>,
-	getVariableValue(variableName: string, input: string): Promise<string>;
+	getOptions(): Promise<string[] | azdata.CategoryValue[]> | string[] | azdata.CategoryValue[],
```

Comment:
Why are you changing this? This doesn't appear to be necessary and having it sometimes return a promise makes it more annoying for callers to have to figure out what the actual return type is. If you always return a promise then you can just have the ones that don't require async stuff just resolve immediately and that's a perfectly fine option.

Reply:
callers don't have to figure anything out as a caller can simply await. The implementers get the option of choosing either return type. See call in modelViewUtils where nothing special is being done.

Reply:
You can - but why? As far as I can tell this isn't needed currently.

Reply:
callers don't have to figure anything out as a caller can simply await. The implementers get the option of choosing either return type. Basically, you are suggesting that implementers just return a Promise.resolve of value that they wish to return then by waiting on a function that returns a simple value and not a promise, the language compiler does the wrapping of Promise.resolve for us. So, by adding that additional return time we are making the code cleaner.

Reply:
My point is that currently I don't see any implementation that needs to return a non-Promise so I don't see the point in updating this now.